provider/pkg/provider/deployment_settings.go

```diff
@@ -482,6 +579,8 @@ func getSecretOrStringValue(prop resource.PropertyValue) string {
 	switch prop.V.(type) {
 	case *resource.Secret:
 		return prop.SecretValue().Element.StringValue()
+	case nil:
```

Comment:
I don't think the type would ever be `nil`?

Reply:
So this was actually a weird bug that I still don't quite understand, but sometimes it happens to be nil, it took me a couple hours to track down a panic to this, because I couldn't believe it. I added this dummy return hoping to figure out how it happens but nothing broke with the "" return at all, so I figured I'll leave it be. I can file an issue to track it down if you think it's worth it